### PR TITLE
add new faculties in the Software Engineering Team of Zhejiang University

### PR DESCRIPTION
**Pull request by baolingfeng**
…sity

# Contributing to CSrankings

Thanks for contributing to CSrankings! Please read and indicate you agree with **all** these guidelines to getting your pull request accepted. Note that pull requests may take some time to get merged (up to three months, as processing has moved to a quarterly cadence).

**If you find CSrankings useful, [please consider becoming a sponsor](https://github.com/sponsors/csrankings).**

**NOTE:** IF YOU DO NOT FOLLOW THE STEPS BELOW, YOUR COMMIT MAY BE SUMMARILY REJECTED.

1. _Use a reasonable title that explains what the PR corresponds to (as in, not "Update csrankings-x.csv")._

1. _Do not modify any files except `csrankings-[a-z].csv` or (if needed) `country-info.csv`._

1. _Do not use Excel to edit any .csv files; Excel incorrectly tries to
convert some Google Scholar entries to formulas, corrupting the
database. Use a text editor like emacs or NotePad instead._

1. _Insert new faculty **in alphabetical order** (not at the end) in the appropriate `csrankings-[a-z].csv` files._ **Do not modify `csrankings.csv`, which is auto-generated.**

1. _Check to make sure that you have no spaces after commas, or any missing fields._

1. _Check to make sure the home page is correct._

1. _Make sure the Google Scholar IDs are just the alphanumeric identifier (not a URL or with `&hl=en`)._

1. _Check to make sure the name corresponds to the DBLP entry (look it up at http://dblp.org)._

1. _If a faculty member is not in a CS department or similar, include a comment explaining how they meet the inclusion criteria (see below)._

1. _Read and check **all** the boxes below by filling them in with an X._

**Inclusion criteria**

- [ ] Make sure that any faculty you add meet the inclusion
criteria. Eligible faculty include only full-time, tenure-track research
faculty members on a given campus who can *solely* advise PhD students in
Computer Science. Faculty not in a CS department or similar who can
advise PhD students in CS can be included regardless of their home
department. Faculty should also have a 75%+ time appointment (check
`old/industry.csv` for faculty who are now more than 25% in industry).

**Updating an affiliation or home page**

- [ ] Update affiliations, home pages, and Google Scholar entries by modifying `csrankings-[a-z].csv`. For the Google Scholar entry, just use the alphanumeric identifier in the middle of the URL. If none is there, put `NOSCHOLARPAGE`.

**Adding one or more faculty members (including an entire department)**

- [ ] If the department is not yet listed in CSrankings, the entire CS faculty needs to be added (not just one faculty member).

- [ ] Enter each faculty member's [DBLP](http://dblp.org) name, home page, and Google Scholar entry (just the alphanumeric identifier, not the whole URL) by modifying `csrankings-[a-z].csv` (**the letters correspond to the first letter of the faculty members' names**); include disambiguation suffixes like `0001` as needed. If the faculty entry is currently ambiguous, please do not include them. Send mail to the DBLP maintainers (dblp@dagstuhl.de) with a few publications by a particular faculty member; also, open an issue so that when the DBLP database is updated, that faculty member's information can be added.

- [ ] If DBLP has multiple entries for this person, *all of them need to be listed*. Do not update `dblp-aliases.csv`.

- [ ] If the institution you are adding is not in the US,
update `country-info.csv` and add *all* of the faculty in the CS department.

**(Advanced) Quick contribution via a shallow clone** 

We recommend that you use the GitHub web user interface to make changes. However, it may be more convenient
to clone the repository for larger-scale changes.

However, a full clone of the CSrankings repository is almost 2GB.  To
contribute a change without creating a full local clone of the
CSrankings repo, you can perform a _shallow clone_.  To do so, follow these
steps:

1. Fork the CSrankings repo.  If you have an existing fork, but it is not up to date with the main repository, this technique may not work.  If necessary, delete and re-create your fork to get it up to date.  (Do **not** delete your existing fork if it has unmerged changes you want to preserve!)

1. Do a shallow clone of your fork: `git clone --depth 1 https://github.com/yourusername/CSrankings`.  This will only download the most recent commit, not the full git history.

1. Make your changes on a branch, push them to your clone, and create a pull request on GitHub as usual.

If you want to make another contribution and some time has passed, perform these steps again, creating a fresh fork and shallow clone.

### CODE REVIEW

File: csrankings-x.csv [hunk 1]
```diff
@@ -245,6 +245,7 @@ Xiaowen Zhang,CUNY,http://www.csi.cuny.edu/faculty/ZHANG_XIAOWEN.html,vX1p5wIAAA
 Xiaoxi He,University of Macau,https://www.fst.um.edu.mo/people/xiaoxi/,jkhl0fUAAAAJ
 Xiaoxing Ma,Nanjing University,http://ics.nju.edu.cn/people/xiaoxingma,44WpWR4AAAAJ
 Xiaoxuan Lu 0001,University of Edinburgh,https://christopherlu.github.io,idu78-EAAAAJ
+Xiaoxue Ren,Zhejiang University,https://scholar.google.com.sg/citations?hl=en&user=mdoN_woAAAAJ,mdoN_woAAAAJ
```
Comment (emeryberger) [inline csrankings-x.csv]: Google Scholar page is not suitable as a home page.

File: csrankings-x.csv [hunk 2]
```diff
@@ -291,7 +292,7 @@ Xin Wang 0001,Stony Brook University,https://www.cs.stonybrook.edu/people/facult
 Xin Wang 0002,Fudan University,http://homepage.fudan.edu.cn/xinw2013,4KcxzhoAAAAJ
 Xin Wang 0019,Tsinghua University,http://mn.cs.tsinghua.edu.cn/xinwang,YPOBHYUAAAAJ
 Xin Wang 0061,Univ. of California - Santa Cruz,https://eric-xw.github.io,YjqluE0AAAAJ
-Xin Xia 0001,Monash University,https://xin-xia.github.io,XSZRxOEAAAAJ
+Xin Xia 0001,Zhejiang University,https://xin-xia.github.io,XSZRxOEAAAAJ
```
Comment (emeryberger) [inline csrankings-x.csv]: Page says now at Huawei.

File: csrankings-z.csv [hunk 3]
```diff
@@ -253,6 +254,7 @@ Zhonghong Ou,BUPT,https://scs.bupt.edu.cn/info/1089/1790.htm,aYvRNFYAAAAJ
 Zhongjie Ba,Zhejiang University,https://person.zju.edu.cn/zhongjieba,dO2kc6kAAAAJ
 Zhongmin Cai,Xi'an Jiaotong University,http://gr.xjtu.edu.cn/web/zmcai/english-version,dkp-d0IAAAAJ
 Zhongqing Wang,Soochow University,http://web.suda.edu.cn/wangzq,IvcHeS8AAAAJ
+Zhongxin Liu,Zhejiang University,https://person.zju.edu.cn/en/liuzhongxin,U_yKZbYAAAAJ
```
Comment (emeryberger) [inline csrankings-z.csv:256]: Web page yields a 502 error.

**Review by emeryberger — CHANGES_REQUESTED** [review]

File: csrankings-x.csv (hunk 1 above)
Comment (baolingfeng) [reply csrankings-x.csv]: I have changed her home page in the new commit

File: csrankings-x.csv (hunk 2 above)
Comment (baolingfeng) [reply csrankings-x.csv]: change his institution to Huawei

File: csrankings-z.csv (hunk 3 above)
Comment (baolingfeng) [reply csrankings-z.csv:256]: I can access his home page now. Could you try it again? Thank you very much.

File: csrankings-x.csv [hunk 4]
```diff
@@ -292,7 +292,7 @@ Xin Wang 0001,Stony Brook University,https://www.cs.stonybrook.edu/people/facult
 Xin Wang 0002,Fudan University,http://homepage.fudan.edu.cn/xinw2013,4KcxzhoAAAAJ
 Xin Wang 0019,Tsinghua University,http://mn.cs.tsinghua.edu.cn/xinwang,YPOBHYUAAAAJ
 Xin Wang 0061,Univ. of California - Santa Cruz,https://eric-xw.github.io,YjqluE0AAAAJ
-Xin Xia 0001,Zhejiang University,https://xin-xia.github.io,XSZRxOEAAAAJ
+Xin Xia 0001,Huawei,https://xin-xia.github.io,XSZRxOEAAAAJ
```
Comment (emeryberger) [inline csrankings-x.csv]: Industry affiliations need to move the record from CSrankings-*.csv and move to `old/industry.csv`.

**Review by emeryberger — CHANGES_REQUESTED** [review]

File: csrankings-x.csv (hunk 4 above)
Comment (baolingfeng) [reply csrankings-x.csv]: Thank you for the reminder. I have move Xin to industry.

Comment (baolingfeng) [reply csrankings-x.csv]: Is this change OK now?

File: csrankings-z.csv [hunk 5]
```diff
@@ -202,6 +202,7 @@ Zhiming Xu,Harbin Institute of Technology,http://homepage.hit.edu.cn/xuzhiming,C
 Zhiming Zhao,University of Amsterdam,https://staff.fnwi.uva.nl/z.zhao/,Pjiw1NUAAAAJ
 Zhipeng Cai,Georgia State University,https://grid.cs.gsu.edu/zcai,tq-LVzIAAAAJ
 Zhipeng Lü,HUST,http://faculty.hust.edu.cn/lzp/en/index.htm,NOSCHOLARPAGE
+Zhipeng Gao,Zhejiang University,https://zpgao.github.io,80_-44AAAAAJ
```
Comment (emeryberger) [inline csrankings-z.csv]: Disambiguation page. Must be removed. Please read directions about reporting to dblp@dblp.org.

**Review by emeryberger — CHANGES_REQUESTED** [review]

File: csrankings-z.csv (hunk 5 above)
Comment (baolingfeng) [reply csrankings-z.csv]: I removed him. Thank you very much!

**Comment by baolingfeng** [comment]
The Commit Validation check was not successful. How can I fix this issue? Could you give any suggestions? Thank you very much.

File: csrankings-z.csv (hunk 5 above)
Comment (emeryberger) [reply csrankings-z.csv]: Still not removed.

**Review by emeryberger — CHANGES_REQUESTED** [review]

File: csrankings-z.csv (hunk 5 above)
Comment (baolingfeng) [reply csrankings-z.csv]: Sorry, I did not notice that my change was not pushed. I have submitted my commits now. Please help check them. Thank you very much!

File: old/industry.csv [hunk 6]
```diff
@@ -195,4 +195,5 @@ Daniel Wigdor,University of Toronto,http://www.dgp.toronto.edu/~dwigdor,x3hvdTsA
 Frank Dellaert,Georgia Institute of Technology,https://www.cc.gatech.edu/~dellaert,ZxXBaswAAAAJ
 Sanja Fidler,University of Toronto,http://www.cs.utoronto.ca/~fidler,CUlqK5EAAAAJ
 Jyh-Ming Lien,George Mason University,http://cs.gmu.edu/~jmlien,EnrEVb0AAAAJ,iRobot
+Xin Xia 0001,Huawei,https://xin-xia.github.io,XSZRxOEAAAAJ
```
Comment (emeryberger) [inline old/industry.csv]: Please follow the format of other entries - keep the original affiliation and put the new affiliation at the end.

**Review by emeryberger — CHANGES_REQUESTED** [review]

File: old/industry.csv (hunk 6 above)
Comment (baolingfeng) [reply old/industry.csv]: Done. Thank you very much！

**Pull request merged**